src/audio/eq_fir/eq_fir.c

```diff
@@ -185,7 +185,7 @@ static void eq_fir_s16_passthrough(struct fir_state_32x16 fir[],
 				   struct audio_stream *sink,
 				   int frames, int nch)
 {
-	audio_stream_copy_s16(source, sink, frames * nch);
+	audio_stream_copy_s16(source, 0, sink, 0, frames * nch);
```

Comment:
would it be better to introduce the offset into the source/sink structures ? This means it can be more easily used by other users without having to modify further APIs ?

Reply:
Yes, modify if API is main drawback of this solution however solution with offsets has drawback with resetting offset. It may be cleared:
a) by user at the end of routine - more lines of code and places to create bug.
b) in produce/consume but it is implicit solution.
Moreover *copy*() has been used in twice in FIR, IIR, SRC - in sum 6 times, so it's not very common function.
Another usage is within pcm_converter, but right now pcm_converter is used in 2 places DAI and HOST. So it's not too late for changes.

Reply:
@lgirdwood @ktrzcinx I think that passing offsets as arguments is less error-prone and if this function is not used that much in FW yet it's ok to change it now